src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java

```diff
@@ -280,55 +280,84 @@ else if (tsDiff >= (-(1L << 30) & 0xFFFFFFFFL))
             int bMinSeqNum = b.getMinSeen(), aMaxSeqNum = a.getMaxSeen();
             int snDiff = (bMinSeqNum - aMaxSeqNum) & 0xFFFF;
 
-            if (bFirstSeqNum != -1 || aLastSeqNum != -1)
+            boolean guessed = false;
+
+            // We have some codecs (VPX) for which we can inspect the payload
+            // to deduce the start/end of a frame. This has an impact on the
+            // distances between the frames that allow us to guess frame
+            // boundaries.
+            boolean withFrameBoundaries =
+                a.supportsFrameBoundaries() && b.supportsFrameBoundaries();
+
+            if (withFrameBoundaries)
             {
-                if (snDiff == 2)
+                if (bFirstSeqNum != -1 || aLastSeqNum != -1)
                 {
-                    if (aLastSeqNum == -1)
+                    if (snDiff == 2 || snDiff == (-2 & 0xFFFF))
```

Comment:
What is the idea behind -2 here? It seems to me that if the diff is -2 then "a" comes after "b", and we cannot apply the logic (setting a's end for example). What am I missing?